src/python/WMCore/MicroService/Unified/RSEQuotas.py

```diff
@@ -0,0 +1,204 @@
+"""
+Non thread-safe object which provides all the RSE/PNN information
+required for automatic data placement.
+It can also communicate with other data management tools, like
+Detox, Rucio and PhEDEx.
+"""
+from __future__ import division, print_function
+
+from WMCore.MicroService.Unified.Common import getDetoxQuota
+from WMCore.MicroService.Unified.Common import getMSLogger
+
+
+class RSEQuotas(object):
+    """
+    Class which represents a list of RSEs, their quota and
+    their storage usage
+    """
+
+    def __init__(self, detoxUrl, dataAcct, quotaFraction, **kwargs):
+        """
+        Executes a basic setup, including proper logging.
+        :param detoxUrl: string with the detox url (to fetch the quota)
+        :param dataAcct: string with either the Rucio account or PhEDEx group name
+        :param quotaFraction: float point number representing the fraction of the quota
+        :param kwargs: the supported keyword arguments are:
+          minimumThreshold: integer value defining the minimum available space required
+          useRucio: boolean flag used to decide between Rucio and PhEDEx data management
+          verbose: logger verbosity
+          logger: logger object
+        """
+        self.detoxUrl = detoxUrl
+        self.dataAcct = dataAcct
+        self.quotaFraction = quotaFraction
+
+        self.minimumSpace = kwargs["minimumThreshold"]
+        self.useRucio = kwargs.get("useRucio", False)
+        self.logger = getMSLogger(kwargs.get("verbose"), kwargs.get("logger"))
+
+        self.nodeUsage = {}
+        self.availableRSEs = set()
+        self.outOfSpaceNodes = set()
+
+    def __str__(self):
+        """
+        Write out useful information for this object
+        :return: a stringified dictionary
+        """
+        res = {'detoxUrl': self.detoxUrl, 'dataAcct': self.dataAcct,
+               'useRucio': self.useRucio, 'quotaFraction': self.quotaFraction,
+               'minimumSpace': self.minimumSpace}
+        return str(res)
```

Comment:
Never use str for getting string representation of the dict. You should always use json.dump. The former will produce improper json and we had so much headache to catch such errors.

Reply:
It goes to a plain text log anyways. But it's a valid concern and I will get it updated.

Reply:
Valentin, I'm keeping this method override as is because if I have a set object in there, json.dumps fails to serialize.
And if someone wants to get a json representation of this data, then we should implement the proper method. The `__str__` is mostly a human readable string.